Test failures appearing in JDK 25

Test failures appearing in JDK 25 by stricter enforcement of networking protocols and resource management in the recent JDK.

- An unexpected "Premature EOF" (End of File) exception
- javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)

-----
```
java.io.IOException: Premature EOF
	at java.base/sun.net.www.MeteredStream.justRead(MeteredStream.java:58)
	at java.base/sun.net.www.MeteredStream.read(MeteredStream.java:135)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:119)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(HttpURLConnection.java:3308)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:279)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:322)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:186)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:183)
	at java.base/java.io.BufferedReader.fill(BufferedReader.java:166)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:333)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:400)
	at org.glassfish.grizzly.http.server@4.1.0-SNAPSHOT/org.glassfish.grizzly.http.server.NetworkListenerTest.testGracefulShutdownGracePeriodExpired(NetworkListenerTest.java:244)
```
Since JDK 24 GA, if EOF is encountered during BufferedReader#readLine execution, an IOException may be thrown instead of null.

- https://github.com/openjdk/jdk/blob/jdk-24-ga/src/java.base/share/classes/sun/net/www/MeteredStream.java
- https://github.com/openjdk/jdk/blob/6705a9255d28f351950e7fbca9d05e73942a4e27/src/java.base/share/classes/sun/net/www/MeteredStream.java#L57-L59
- https://bugs.openjdk.org/browse/JDK-8335135 - https://github.com/openjdk/jdk/commit/b6ffb442acb4a222f017868433eff213d9b84ed8



```
[ERROR]   AlpnSupportTest.sslToConnectionMapClearTest:53 » Timeout
[ERROR] org.glassfish.grizzly.http2.AlpnSupportTest.sslToConnectionMapClearTest -- Time elapsed: 10.08 s <<< ERROR!
java.util.concurrent.TimeoutException
	at org.glassfish.grizzly.impl.SafeFutureImpl$Sync.innerGet(SafeFutureImpl.java:324)
	at org.glassfish.grizzly.impl.SafeFutureImpl.get(SafeFutureImpl.java:237)
	at org.glassfish.grizzly.http2.AlpnSupportTest.sslToConnectionMapClearTest(AlpnSupportTest.java:53)
```
Internally, javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate) is occurring.

Consideration should be given to avoiding this error in JDK 25 as well.

--- a/modules/grizzly/src/main/resources/org/glassfish/grizzly/localization/log.properties
+++ b/modules/grizzly/src/main/resources/org/glassfish/grizzly/localization/log.properties
@@ -1,4 +1,5 @@
 #
+# Copyright (c) 2025 Contributors to the Eclipse Foundation.
 # Copyright (c) 2010, 2017 Oracle and/or its affiliates. All rights reserved.
 #
 # This program and the accompanying materials are made available under the

--- a/modules/http-server/src/test/java/org/glassfish/grizzly/http/server/NetworkListenerTest.java
+++ b/modules/http-server/src/test/java/org/glassfish/grizzly/http/server/NetworkListenerTest.java
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
  * Copyright (c) 2010, 2020 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -23,6 +24,7 @@ import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
 import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.HttpURLConnection;
 import java.net.URL;
@@ -241,7 +243,15 @@ public class NetworkListenerTest {
             Future<HttpServer> gracefulFuture = server.shutdown(1, TimeUnit.SECONDS);
 
             final BufferedReader reader = new BufferedReader(new InputStreamReader(c.getInputStream(), Charsets.UTF8_CHARSET));
-            final String content = reader.readLine();
+            String content;
+            try {
+                // this should result in null in older JDKs(~ JDK 23) or IOException on newer JDKs(JDK 24 GA ~) as the grace period expires
+                content = reader.readLine();
+            } catch (IOException e) {
+                // starting with JDK-24-GA, when EOF is encountered during BufferedReader#readLine() execution,
+                // "java.io.IOException: Premature EOF" is thrown instead of returning null
+                content = null;
+            }
             assertNull(content);
 
             assertNotNull(gracefulFuture.get(5, TimeUnit.SECONDS));

--- a/modules/http/src/main/java/org/glassfish/grizzly/http/util/Parameters.java
+++ b/modules/http/src/main/java/org/glassfish/grizzly/http/util/Parameters.java
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
  * Copyright (c) 2010, 2020 Oracle and/or its affiliates. All rights reserved.
  * Copyright 2004 The Apache Software Foundation
  *

--- a/modules/http/src/test/java/org/glassfish/grizzly/http/ParametersTest.java
+++ b/modules/http/src/test/java/org/glassfish/grizzly/http/ParametersTest.java
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
  * Copyright (c) 2011, 2020 Oracle and/or its affiliates. All rights reserved.
  * Copyright 2004 The Apache Software Foundation
  *

--- a/modules/http2/src/test/java/org/glassfish/grizzly/http2/AbstractHttp2Test.java
+++ b/modules/http2/src/test/java/org/glassfish/grizzly/http2/AbstractHttp2Test.java
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
  * Copyright (c) 2010, 2020 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -154,11 +155,11 @@ public abstract class AbstractHttp2Test {
                     SSLContextConfigurator sslContextConfigurator = createSSLContextConfigurator();
                     serverSSLEngineConfigurator = new SSLEngineConfigurator(sslContextConfigurator.createSSLContext(true), false, false, false);
 
-                    serverSSLEngineConfigurator.setEnabledCipherSuites(new String[] { "TLS_RSA_WITH_AES_256_CBC_SHA" });
+                    serverSSLEngineConfigurator.setEnabledCipherSuites(new String[] { "TLS_AES_128_GCM_SHA256" });
 
                     clientSSLEngineConfigurator = new SSLEngineConfigurator(sslContextConfigurator.createSSLContext(true), true, false, false);
 
-                    clientSSLEngineConfigurator.setEnabledCipherSuites(new String[] { "TLS_RSA_WITH_AES_256_CBC_SHA" });
+                    clientSSLEngineConfigurator.setEnabledCipherSuites(new String[] { "TLS_AES_128_GCM_SHA256" });
                 }
             }
         }

